Rename "computedCSSPadding()" to "resolveLengthPercentage()"

smfr opened the pull request:
#### 4142d950a6e7766786f1f3e2bd4650e3228628ec
<pre>
Rename &quot;computedCSSPadding()&quot; to &quot;resolveLengthPercentage()&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=273672">https://bugs.webkit.org/show_bug.cgi?id=273672</a>
<a href="https://rdar.apple.com/problem/127473476">rdar://problem/127473476</a>

Reviewed by Tim Nguyen.

`computedCSSPadding()` was called for margin values in `RenderBox::marginBoxRect()`,
so is clearly misnamed. It&apos;s really resolving the CSS `&lt;length-percentage&gt;`[1] values
which use the container&apos;s logical width (which is true for margins and padding)
so name it thusly.

[1] <a href="https://drafts.csswg.org/css-values-4/#mixed-percentages">https://drafts.csswg.org/css-values-4/#mixed-percentages</a>

* Source/WebCore/rendering/RenderBoxInlines.h:
(WebCore::RenderBox::marginBoxRect const):
* Source/WebCore/rendering/RenderBoxModelObject.h:
* Source/WebCore/rendering/RenderBoxModelObjectInlines.h:
(WebCore::RenderBoxModelObject::computedCSSPaddingAfter const):
(WebCore::RenderBoxModelObject::computedCSSPaddingBefore const):
(WebCore::RenderBoxModelObject::computedCSSPaddingBottom const):
(WebCore::RenderBoxModelObject::computedCSSPaddingEnd const):
(WebCore::RenderBoxModelObject::computedCSSPaddingLeft const):
(WebCore::RenderBoxModelObject::computedCSSPaddingRight const):
(WebCore::RenderBoxModelObject::computedCSSPaddingStart const):
(WebCore::RenderBoxModelObject::computedCSSPaddingTop const):
(WebCore::RenderBoxModelObject::resolveLengthPercentageUsingContainerLogicalWidth const):
(WebCore::RenderBoxModelObject::computedCSSPadding const): Deleted.

Canonical link: <a href="https://commits.webkit.org/278335@main">https://commits.webkit.org/278335@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/29f0a875a1dd196bb1d1ed7e1de60b0cb43974bf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50226 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29518 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/2521 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/53485 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/916 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/52529 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/35746 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/536 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/40982 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52325 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27188 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43227 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22080 "Found 1 new API test failure: /WPE/TestResources:/webkit/WebKitWebView/sync-request-on-max-conns (failure)") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/24609 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/479 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/8604 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/46596 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/540 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55070 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25322 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/475 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/48380 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/26583 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43415 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47402 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11021 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/27447 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/26315 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/rendering/RenderBoxModelObjectInlines.h:
```diff
@@ -88,7 +88,7 @@ inline RectEdges<LayoutUnit> RenderBoxModelObject::borderWidths() const
 }
 
-inline LayoutUnit RenderBoxModelObject::computedCSSPadding(const Length& padding) const
+inline LayoutUnit RenderBoxModelObject::resolveLengthPercentage(const Length& padding) const
```
The argument should be renamed as well.

fantasai reviewed (changes requested):
The new name doesn't make it clear that this resolves against the containing block logical width. The current name might be suboptimal, but at least clearly ties it to padding behavior.

Maybe `resolveMarginPaddingPercentage` or something like that could work.

fantasai reviewed (approved):
Assuming updated commit message

smfr committed:
Rename "computedCSSPadding()" to "resolveLengthPercentage()"
https://bugs.webkit.org/show_bug.cgi?id=273672
rdar://problem/127473476

Reviewed by Tim Nguyen.

`computedCSSPadding()` was called for margin values in `RenderBox::marginBoxRect()`,
so is clearly misnamed. It's really resolving the CSS `<length-percentage>`[1] values
which use the container's logical width (which is true for margins and padding)
so name it thusly.

[1] https://drafts.csswg.org/css-values-4/#mixed-percentages

* Source/WebCore/rendering/RenderBoxInlines.h:
(WebCore::RenderBox::marginBoxRect const):
* Source/WebCore/rendering/RenderBoxModelObject.h:
* Source/WebCore/rendering/RenderBoxModelObjectInlines.h:
(WebCore::RenderBoxModelObject::computedCSSPaddingAfter const):
(WebCore::RenderBoxModelObject::computedCSSPaddingBefore const):
(WebCore::RenderBoxModelObject::computedCSSPaddingBottom const):
(WebCore::RenderBoxModelObject::computedCSSPaddingEnd const):
(WebCore::RenderBoxModelObject::computedCSSPaddingLeft const):
(WebCore::RenderBoxModelObject::computedCSSPaddingRight const):
(WebCore::RenderBoxModelObject::computedCSSPaddingStart const):
(WebCore::RenderBoxModelObject::computedCSSPaddingTop const):
(WebCore::RenderBoxModelObject::resolveLengthPercentageUsingContainerLogicalWidth const):
(WebCore::RenderBoxModelObject::computedCSSPadding const): Deleted.

Canonical link: https://commits.webkit.org/278335@main
